uniffi/src/ffi/foreigncallbacks.rs

```diff
@@ -131,8 +132,12 @@ use std::sync::atomic::{AtomicUsize, Ordering};
 ///   passing individual arguments to the user's callback.
 /// * `buf_ptr` is a pointer to where the resulting buffer will be written. UniFFI will allocate a
 ///   buffer to write the result into.
-/// * A callback returns `-1` if an unexpected error happens, `0` if the call succeeded without a
-///   return value or `1` if data was written into the output buffer.
+/// * A callback returns:
+///    - `-2` An error occured that was serialized to buf_ptr
+///    - `-1` An unexpected error ocurred
+///    - `0` is a deprecated way to signal that if the call succeeded, but did not modify buf_ptr
+///    - `1` If the call succeeded.  For non-void functions the return value should be serialized
```

Comment:
but this confuses me a little given the above - so maybe I'm wrong about what `Ok(())` does?